pcmanfm/desktopwindow.cpp

```diff
@@ -199,6 +207,206 @@ DesktopWindow::~DesktopWindow() {
+        createTrash();
+    }
+    else {
+        if (trashMonitor_) {
+            g_signal_handlers_disconnect_by_func(trashMonitor_, (gpointer)G_CALLBACK(onTrashChanged), this);
+            g_object_unref(trashMonitor_);
+            trashMonitor_ = nullptr;
+        }
+        if(!firstCall) {
+            QString trash = XdgDir::readDesktopDir() + QLatin1String("/trash-can.desktop");
+            if(QFile::exists(trash)) {
+                paths.push_back(Fm::FilePath::fromLocalPath(trash.toStdString().c_str()));
+            }
+        }
+    }
+    // Home
+    if(ds.contains(QLatin1String("Home"))) {
+        createHomeShortcut();
+    }
+    else if(!firstCall) {
+        QString home = XdgDir::readDesktopDir() + QLatin1String("/user-home.desktop");
+        if(QFile::exists(home)) {
+            paths.push_back(Fm::FilePath::fromLocalPath(home.toStdString().c_str()));
+        }
+    }
+    // Computer
+    if(ds.contains(QLatin1String("Computer"))) {
+        createComputerShortcut();
+    }
+    else if(!firstCall) {
+        QString computer = XdgDir::readDesktopDir() + QLatin1String("/computer.desktop");
+        if(QFile::exists(computer)) {
+            paths.push_back(Fm::FilePath::fromLocalPath(computer.toStdString().c_str()));
+        }
+    }
+    // Network
+    if(ds.contains(QLatin1String("Network"))) {
+        createNetworkShortcut();
+    }
+    else if(!firstCall) {
+        QString network = XdgDir::readDesktopDir() + QLatin1String("/network.desktop");
+        if(QFile::exists(network)) {
+            paths.push_back(Fm::FilePath::fromLocalPath(network.toStdString().c_str()));
+        }
+    }
+
+    // WARNING: QFile::remove() is not compatible with libfm-qt and should not be used.
+    if(!paths.empty()) {
+        Fm::FileOperation::deleteFiles(paths, false);
+    }
+
+    firstCall = false; // desktop is created
+}
+
+void DesktopWindow::createTrashShortcut(int items) {
+    GKeyFile* kf = g_key_file_new();
+    g_key_file_set_string(kf, "Desktop Entry", "Type", "Application");
+    g_key_file_set_string(kf, "Desktop Entry", "Exec", "pcmanfm-qt trash:///");
+    // icon
+    const char* icon_name = items > 0 ? "user-trash-full" : "user-trash";
+    g_key_file_set_string(kf, "Desktop Entry", "Icon", icon_name);
+    // name
+    QString name;
+    if(items > 0) {
+        if (items == 1){
+            name = tr("Trash (One item)");
+        }
+        else {
+            name = tr("Trash (%1 items)").arg(items);
+        }
+    }
+    else {
+        name = tr("Trash (Empty)");
+    }
+    g_key_file_set_string(kf, "Desktop Entry", "Name", name.toLatin1().constData());
+
+    auto path = Fm::FilePath::fromLocalPath(XdgDir::readDesktopDir().toStdString().c_str()).localPath();
+    auto trash_can = Fm::CStrPtr{g_build_filename(path.get(), "trash-can.desktop", nullptr)};
+    g_key_file_save_to_file(kf, trash_can.get(), nullptr);
+    g_key_file_free(kf);
+}
+
+void DesktopWindow::createHomeShortcut() {
```

Comment:
Only a idea - if we pre-create these four files and put them somewhere™ (/usr/share/applications/lxqt/$foo.desktop or so) we could just copy them - the adventage would be: We could translate the desktop icons. If the idea sound nuts please ignore it.

Reply:
You don't need to create them; they will be created automatically if the new config key (`DesktopShortcuts` under the `[Desktop]` section) has them -- and it will if the checkboxes are checked. Distros can make it like this:
```
...
DesktopShortcuts=Home, Trash, Computer, Network
...
```
They will also be removed automatically if checkboxes are unchecked. So, the user doesn't need to worry about desktop entries.

You'll see it when you test it.

Reply:
As for localization, the desktop entries will be created according to new translatable strings. I've done it for Trash but forgot the other items. Good catch! Will add a commit very soon.

Reply:
Done! Now all names (what users see) are localized.

Even if a bad distro does the bad thing and adds dekstop entries to Desktop, they will be updated/overwritten once pcmanfm-qt starts -- in a localized way.